### PR TITLE
Use gh-pages package for deployment

### PR DESCRIPTION
**Pull request by rekmarks**
This PR updates our deployment script to use the [`gh-pages` package](https://www.npmjs.com/package/gh-pages) for deployment, and removes a lot of complexity from it.

I tested deployment once from `main`, and `gh-pages` returned successfully, although without updating anything because the website content hadn't changed.

I decided to keep `deploy.sh` for the dirty working tree check and the fancy commit message. Otherwise it could've easily fit in a `package.json` script.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "bugs": {
     "url": "https://github.com/MetaMask/test-dapp/issues"
   },
-  "homepage": "https://github.com/MetaMask/test-dapp#readme",
+  "homepage": "https://metamask.github.io/test-dapp",
```
Comment (rekmarks) [inline package.json:32]: Pointing to the deployed website is required for `gh-pages` to work correctly in at least some instances, and this seems like a good change regardless.

File: deploy.sh [hunk 2]
```diff
@@ -3,14 +3,11 @@
...
 
-readonly __SCRIPT_NAME__="${0##*/}"
-readonly __SEE_HELP_MESSAGE__="See '${__SCRIPT_NAME__} --help' for more information."
-
-GH_REMOTE='origin'
-SOURCE_BRANCH='main'
-DEPLOY_BRANCH='gh-pages'
-WEBSITE_DIR_PATH='dist'
-RETURN_PATH='..'
+readonly GH_REMOTE="origin"
```
Comment (Gudahtt) [inline deploy.sh:6]: Hmm. I had found it useful to specify this via the CLI - it's how I QA'd previous changes to the deploy script for example.

File: deploy.sh [hunk 3]
```diff
@@ -51,78 +36,28 @@ function get_shorthash {
...
 
   local shorthash
   shorthash=$(get_shorthash)
 
-  local remote_url
-  remote_url=$(git remote get-url "${GH_REMOTE}")
+  local commitMessage="Update using ${SOURCE_BRANCH}/${shorthash}"
 
-  cd ${WEBSITE_DIR_PATH} || abort "Failed to navigate to website directory"
+  node "$GH_PAGES_BIN_PATH" --dist "$WEBSITE_DIR_PATH" --message "$commitMessage" || abort "gh-pages failed to deploy"
```
Comment (Gudahtt) [inline deploy.sh:49]: Might it be simpler here to use `yarn gh-pages`?

```suggestion
  yarn gh-pages --dist "$WEBSITE_DIR_PATH" --message "$commitMessage" || abort "gh-pages failed to deploy"
```

Comment (rekmarks) [reply deploy.sh:49]: Ah! I forgot this was a possibility. Pushing this up.

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**